libs/wingc/src/jsify.rs

```diff
@@ -1961,3 +1984,51 @@ fn jsify_type_name(t: &Vec<Symbol>, phase: Phase) -> String {
 		p.join(".")
 	}
 }
+
+fn lookup_span(span: &WingSpan, source_path: &Path) -> String {
+	let source = std::fs::read_to_string(source_path)
+		.expect(format!("Failed to read source file \"{}\"", source_path.display()).as_str());
+	let lines = source.lines().collect_vec();
+
+	let start_line = span.start.line as usize;
+	let end_line = span.end.line as usize;
+
+	let start_col = span.start.col as usize;
+	let end_col = span.end.col as usize;
+
+	let mut result = String::new();
+
+	if start_line == end_line {
+		result.push_str(&lines[start_line][start_col..end_col]);
+	} else {
+		result.push_str(&lines[start_line][start_col..]);
+		result.push('\n');
+
+		for line in lines[start_line + 1..end_line].iter() {
+			result.push_str(line);
+			result.push('\n');
+		}
+
+		result.push_str(&lines[end_line][..end_col]);
+	}
+
+	result
+}
+
+fn escape_javascript_string(s: &str) -> String {
```

Comment:
maybe a couple of a small unit tests for these helper functions?

Reply:
Done